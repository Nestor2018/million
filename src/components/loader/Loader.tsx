import React from 'react';
import { StyleSheet, ActivityIndicator, ActivityIndicatorProps } from 'react-native';

interface LoaderProps extends ActivityIndicatorProps { }

/**
 * Componente de Loader (Muestra un indicador de carga cuando se está cargando información en la pantalla)
 * @param {LoaderProps} props - Propiedades para personalizar el loader
 * @returns {JSX.Element} El componente renderizado
 */
const Loader: React.FC<LoaderProps> = ({ color = 'white', size = 'large', style, ...rest }) => (
  <ActivityIndicator color={color} size={size} style={[styles.loader, style]} {...rest} testID="loader" />
);

const styles = StyleSheet.create({
  loader: {
    margin: 20,
  },
});

export default Loader;
