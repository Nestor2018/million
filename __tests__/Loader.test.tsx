import React, { memo } from 'react';
import { render } from '@testing-library/react-native';
import Loader from "../src/components/loader/Loader"

describe('Loader Component', () => {
  const MemoizedLoader = memo(Loader);

  it('should render the Loader component', () => {
    const { getByTestId } = render(<Loader />);
    const loader = getByTestId('loader');

    expect(loader).toBeTruthy();
  });

  it('should apply custom color and size', () => {
    const { getByTestId } = render(<Loader color="red" size="small" />);
    const loader = getByTestId('loader');

    // Verifica que las props se apliquen correctamente
    expect(loader.props.color).toBe('red');
    expect(loader.props.size).toBe('small');
  });

  it('should combine default styles with custom styles', () => {
    const customStyle = { margin: 50 };
    const { getByTestId } = render(<Loader style={customStyle} />);
    const loader = getByTestId('loader');

    expect(loader.props.style).toEqual(expect.arrayContaining([customStyle]));
  });

  it('matches snapshot', () => {
    const tree = render(<Loader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('is accessible with proper accessibility labels', () => {
    const { getByTestId } = render(<Loader accessibilityLabel="Loading..." />);
    const loader = getByTestId('loader');

    // Verifica que el componente tenga un label de accesibilidad
    expect(loader.props.accessibilityLabel).toBe('Loading...');
  });

  it('should pass additional props to ActivityIndicator', () => {
    const customProps = { animating: false, hidesWhenStopped: true };
    const { getByTestId } = render(<Loader {...customProps} />);
    const loader = getByTestId('loader');

    // Verifica que las props adicionales se apliquen correctamente
    expect(loader.props.animating).toBe(false);
    expect(loader.props.hidesWhenStopped).toBe(true);
  });

  it('does not re-render when props do not change', () => {
    const { rerender, getByTestId } = render(<MemoizedLoader />);
    const initialRender = getByTestId('loader');

    // Re-renderiza el componente con las mismas props
    rerender(<MemoizedLoader />);
    const afterRerender = getByTestId('loader');

    // Verifica que el componente no haya cambiado
    expect(afterRerender).toBe(initialRender);
  });
});
