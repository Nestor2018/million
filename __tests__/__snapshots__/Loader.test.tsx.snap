// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component matches snapshot 1`] = `
<ActivityIndicator
  accessibilityLabel="Loading..."
  color="white"
  size="large"
  style={
    [
      undefined,
      undefined,
    ]
  }
  testID="loader"
/>
`;
